docs/source/conf.py

```diff
@@ -3378,6 +3382,39 @@
     html_title = " ".join((project, version, "documentation"))
     release = version
 
+# Use the linkcode extension to override [SOURCE] links to point
+# to the repo. Use the torch_version variable defined above to
+# determine link
+def linkcode_resolve(domain, info):
+    if domain != "py":
+        return None
+    if not info["module"]:
+        return None
+
+    try:
+        module = __import__(info["module"], fromlist=[""])
+        obj = module
+        for part in info["fullname"].split("."):
+            obj = getattr(obj, part)
+        # Get the source file and line number
+        fn = inspect.getsourcefile(obj)
+        source, lineno = inspect.getsourcelines(obj)
+    except Exception:
+        return None
+
+    # Determine the tag based on the torch_version
+    if RELEASE:
+        # For release versions, format as "vX.Y.Z" for correct path in repo        
+        version_parts = torch_version.split(".")
+        patch_version = version_parts[2].split("+")[0].split("a")[0] # assuming a0 always comes after release version in versions.txt
+        version_path = f"v{version_parts[0]}.{version_parts[1]}.{patch_version}"
+    else:
+        version_path = os.getenv("SHA1", "main")
```

Comment:
Thank Svetlana for the tip, this way docs build would produce the same results for local as well as CI builds
```suggestion
        version_path = torch.version.git_version
```